### DISCUSSION TITLE
call function doc

### DISCUSSION THREAD

**Discussion by oneanime** [spring-projects/spring-ai#303] [Q&A] [answered]
is three a demo about calling function and configure spring.ai.openai.chat.options.tools

**Comment by tzolov** [comment, marked as answer]
Initial support for Function calling is provided with: https://github.com/spring-projects/spring-ai/commit/ffd8222c209967552ea32a265837a48e72076b13
Follow the related documentation: https://docs.spring.io/spring-ai/reference/api/clients/functions/openai-chat-functions.html
Further improvements, extensions and simplifications will come with next (0.9.0) release.

**Comment by radhakrishna67** [comment]
@tzolov , Thanks for the info.
The documentation links are not working.
![image](https://github.com/spring-projects/spring-ai/assets/55708626/f4450f73-7b92-4f56-ac3e-71438f897b06)

**Reply by ThomasVitale** [reply]
You can find the function calling documentation here: https://docs.spring.io/spring-ai/reference/api/functions.html
